File: android/src/main/kotlin/com/emarsys/emarsys_sdk/command/mobileengage/inbox/AddTagCommand.kt
package com.emarsys.emarsys_sdk.command.mobileengage.inbox

import com.emarsys.Emarsys
import com.emarsys.emarsys_sdk.command.EmarsysCommand
import com.emarsys.emarsys_sdk.command.ResultCallback

class AddTagCommand : EmarsysCommand {
    override fun execute(parameters: Map<String, Any?>?, resultCallback: ResultCallback) {
        val messageId = parameters?.get("messageId") as? String
        val tag = parameters?.get("tag") as? String
        messageId ?: throw IllegalArgumentException("messageId must not be null")
        tag ?: throw IllegalArgumentException("tag must not be null")

        Emarsys.messageInbox.addTag(tag, messageId) {
            resultCallback.invoke(null, it)
        }
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        return true
    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }

}

File: ios/Classes/Wrapper/Commands/Inbox/AddTagCommand.swift
//
//  Created by Emarsys on 2021. 08. 19..
//

import EmarsysSDK

class AddTagCommand: EmarsysCommandProtocol {
        
    func execute(arguments: [String : Any]?, resultCallback: @escaping ResultCallback) {
        guard let tag = arguments?["tag"] as? String else {
            resultCallback(["error": "Invalid tag"])
            return
        }
        guard let messageId = arguments?["messageId"] as? String else {
            resultCallback(["error": "Invalid messageId"])
            return
        }
        Emarsys.messageInbox.addTag(tag: tag,
                                    messageId: messageId) { error in
            if let e = error {
                resultCallback(["error": e])
            } else {
                resultCallback(["success": true])
            }
        }
    }
}
